components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -172,7 +172,14 @@ public AbstractAPIManager(String username) throws APIManagementException {
                 int tenantId = getTenantManager().getTenantId(tenantDomainName);
                 this.tenantId = tenantId;
                 this.tenantDomain = tenantDomainName;
-                this.username = tenantUserName;
+                if (APIConstants.SUPER_TENANT_DOMAIN.equalsIgnoreCase(tenantDomainName) &&
+                        !tenantUserName.endsWith(APIConstants.SUPER_TENANT_DOMAIN) &&
```

Comment:
as this is the tenant aware username, we don't need this line I think.

Reply:
+1 yes we don't need to check it.
Thanks.